base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/configuration/ConfigInfo.java

```diff
@@ -148,6 +149,27 @@ public String apply(ConfigInfo info) {
         
         return namesWithProtectionFlag;
     }
+    
+    /**
+     * Checks if there is a protected configuration/comp within the given ConfigInfos objects.
+     * @param infos
+     * @return boolean: True if a protected configuration exists within given infos, False if not
+     */
+    public static Boolean hasProtectedConfigs(Collection<ConfigInfo> infos) {
+        if (infos.isEmpty()) {
```

Comment:
Suggestion: Use infos.stream().anyMatch(<predicate>) where predicate is the test for isProtected. See here for more details on anymatch https://www.geeksforgeeks.org/stream-anymatch-java-examples/. Stream can be a useful tool for situations like this

Reply:
If you choose to implement this suggestion then there is no need for the empty test, but a null check would be useful